### ISSUE TITLE
Compare validation not work

### ISSUE THREAD

**Issue by Bertagit** [yiisoft/yii2#16508]
On a simple ActiveForm to change user password the attributes new_password and repeat_password not are validated also if both have the same value. In the javascript validation a log on $form.data('yiiActiveForm').attributes  show only 2 attribute instead 3 (email, new_password and repeat_password), the attribute lost is new_password and the validation result return always false.

Scenario configuration:

```php
self::SCENARIO_PROFILE => ['email', 'new_password', 'repeat_password']
```

Validations roles:

```php
['new_password', 'safe'],
['repeat_password', 'required', 
       'when' => function($model){
            return !empty($model->new_password);
},
       'whenClient' => "function(attribute, value){
            return $('#user-new_password').val() != '';
        }"
 ],
 ['repeat_password', 'compare', 'compareAttribute' => 'new_password', 'operator' => '=='],
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      |  7.2.7
| Operating system | Ubuntu 18.04

**Comment by Bertagit** [comment]
No more news about this bug? Still not working.
Thanks for the support!

**Comment by samdark** [comment]
Nope.

**Comment by s1lver** [comment]
This bug fixed PR #16523
